odoo/tools/sql.py

```diff
@@ -93,30 +93,42 @@ def to_flush(self) -> Iterator[Field]:
         """ Return an iterator on the fields to flush in the metadata of
         ``self`` and all of its parts.
         """
-        if self.__to_flush is not None:
-            yield self.__to_flush
-        for arg in self.__args:
-            if isinstance(arg, SQL):
-                yield from arg.to_flush
+        stack = [self]
+        while stack:
+            node = stack.pop()
+            if isinstance(node, SQL):
+                if node.__to_flush is not None:
+                    yield node.__to_flush
+                stack.extend(node.__args)
```

Comment:
Because of the order [I wasn't sure](https://github.com/odoo/odoo/compare/4ac38a18ddb8a65e678d3b3c965775a8eab96d44..867465c65218e04c602f8a57023ba884166e87ee), pushed it again anyway.